aristanetworks/ansible-avd
# Fix(eos_validate_state): ANTA Refactor BGP test to now also support direct neighbors and IPv6 AFI

## Change Summary

Refactor `AvdTestBGP` to now also support direct neighbors and IPv6 in addition of peer groups.

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_validate_state` **ANTA**

## Proposed changes
`AvdTestBGP` will now test all activated neighbors and peer groups in the IPv4, IPv6 and EVPN address families of VRF default. This will NOT test BGP neighbors configured in a VRF (coming soon in a separate PR).

## How to test

You can add custom structured_config for BGP under specific nodes:
```yaml
l3leaf:  
  node_groups:
    - group: DC2_BL
      mlag: false
      structured_config:
        router_bgp:
          address_family_ipv4:
            neighbors:
            - ip_address: 123.1.1.10
              activate: true
            - ip_address: 123.1.1.11
              activate: true
          neighbors:
          - ip_address: 123.1.1.10
            remote_as: '1234'
            description: External IPv4 BGP peer
          - ip_address: 123.1.1.11
            remote_as: '1234'
            description: External IPv4 BGP peer
          - ip_address: fd5a:fe45:8831:06c5::a
            remote_as: '12345'
          - ip_address: fd5a:fe45:8831:06c5::b
            remote_as: '12345'
          address_family_ipv6:
            neighbors:
            - ip_address: fd5a:fe45:8831:06c5::a
              activate: true
            - ip_address: fd5a:fe45:8831:06c5::b
              activate: true
```
`eos_validate_state` in ANTA mode will now test the additional activated IPv4, IPv6 neighbors.

Also see `eos_validate_state` molecule scenario.

## Checklist

### User Checklist


- N/A

### Repository Checklist
